src/Styles/Hooks.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace WPML\PB\Cornerstone\Styles;
+
+use WPML\FP\Fns;
+use WPML\FP\Maybe;
+use WPML\LIB\WP\Post;
+use function WPML\FP\partialRight;
+
+class Hooks implements \IWPML_Backend_Action, \IWPML_Frontend_Action, \IWPML_DIC_Action {
+
+	const META_KEY = '_cs_generated_styles';
+
+	/** @var \WPML_PB_Last_Translation_Edit_Mode $lastEditMode */
+	private $lastEditMode;
+
+	/** @var \WPML_Cornerstone_Data_Settings $dataSettings */
+	private $dataSettings;
+
+	/**
+	 * Hooks constructor.
+	 *
+	 * @param \WPML_PB_Last_Translation_Edit_Mode $lastEditMode
+	 * @param \WPML_Cornerstone_Data_Settings     $dataSettings
+	 */
+	public function __construct(
+		\WPML_PB_Last_Translation_Edit_Mode $lastEditMode,
+		\WPML_Cornerstone_Data_Settings $dataSettings
+	) {
+		$this->lastEditMode   = $lastEditMode;
```

Comment:
@strategio I've been thinking about these sort of classes where most of the code it actually about storing the injected objects.

In this case we could do ( if `->filter( $this->shouldInvalidateStyles )` works ):

```php
/* @var callable $shouldInvalidateStyles */
private $shouldInvalidateStyles;

...
$this->shouldInvalidateStyles = Logic::both( [ $dataSettings, 'is_handling_post' ], [ $lastEditMode, 'is_translation_editor' ] );

...

Maybe::of( $postId )
			->filter( $this->shouldInvalidateStyles )
			->map( Post::deleteMeta( Fns::__, self::META_KEY ) );
```

Do you want to try this and see how it looks?

Reply:
Right Bruce, it works nicely :)